flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSink.java

```diff
@@ -0,0 +1,153 @@
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.junit.After;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TestFlinkTableSink extends FlinkCatalogTestBase {
+  private static final String TABLE_NAME = "test_table";
+  private TableEnvironment tEnv;
+  private org.apache.iceberg.Table icebergTable;
+
+  private final FileFormat format;
+  private final boolean isStreamingJob;
+
+  @Parameterized.Parameters(name = "{index}: format={0}, isStreaming={1}, catalogName={2}, baseNamespace={3}")
+  public static Iterable<Object[]> parameters() {
+    List<Object[]> parameters = Lists.newArrayList();
+    for (FileFormat format : new FileFormat[] {FileFormat.ORC, FileFormat.AVRO, FileFormat.PARQUET}) {
+      for (Boolean isStreaming : new Boolean[] {true, false}) {
+        for (Object[] catalogParams : FlinkCatalogTestBase.parameters()) {
+          String catalogName = (String) catalogParams[0];
+          String[] baseNamespace = (String[]) catalogParams[1];
+          parameters.add(new Object[] {format, isStreaming, catalogName, baseNamespace});
+        }
+      }
+    }
+    return parameters;
+  }
+
+  public TestFlinkTableSink(FileFormat format, Boolean isStreamingJob, String catalogName, String[] baseNamespace) {
+    super(catalogName, baseNamespace);
+    this.format = format;
+    this.isStreamingJob = isStreamingJob;
+  }
+
+  @Override
+  protected TableEnvironment getTableEnv() {
+    if (tEnv == null) {
+      synchronized (this) {
+        EnvironmentSettings.Builder settingsBuilder = EnvironmentSettings
+            .newInstance()
+            .useBlinkPlanner();
+        if (isStreamingJob) {
+          settingsBuilder.inStreamingMode();
+          StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+          env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime);
+          env.enableCheckpointing(400);
+          tEnv = StreamTableEnvironment.create(env, settingsBuilder.build());
+        } else {
+          settingsBuilder.inBatchMode();
+          tEnv = TableEnvironment.create(settingsBuilder.build());
+        }
+      }
+    }
+    return tEnv;
+  }
+
+  @Before
+  public void before() {
+    sql("CREATE DATABASE %s", flinkDatabase);
+    sql("USE CATALOG %s", catalogName);
+    sql("USE %s", DATABASE);
+
+    Map<String, String> properties = ImmutableMap.of(TableProperties.DEFAULT_FILE_FORMAT, format.name());
+    this.icebergTable = validationCatalog
```

Comment:
We could use flink DDL to create table here if https://github.com/apache/iceberg/pull/1393 get merged.

Reply:
It was merged!